### PR TITLE
[type: refactor] Add protection when ldapprop is null

### PR DESCRIPTION
**Pull request by QinHaiqi**
// Describe your PR here; eg. Fixes #issueNo

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by AhahaGe** [comment]
How about reduce your commit number, and detail your pr?

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java [hunk 1]
```diff
@@ -47,17 +48,19 @@
...
-        contextSource.setUserDn(ldapProp.getBindDn());
-        contextSource.setPassword(ldapProp.getPassword());
-        contextSource.setPooled(true);
-        Map<String, Object> config = new HashMap<>();
-        config.put("java.naming.ldap.attributes.binary", "objectGUID");
-        config.put("com.sun.jndi.ldap.connect.timeout", String.valueOf(ldapProp.getConnectTimeout()));
-        config.put("com.sun.jndi.ldap.read.timeout", String.valueOf(ldapProp.getReadTimeout()));
-        contextSource.setBaseEnvironmentProperties(config);
-        return contextSource;
+        return Optional.ofNullable(ldapProp).map(item -> {
```
Comment (KevinClair) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java:51]: Parameter `ldapprop` may be null?

**Review by KevinClair — COMMENTED** [review]
Parameter ldapprop may be null?

**Comment by KevinClair** [comment]
By the way, modify your pr's title.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java (hunk 1 above)
Comment (QinHaiqi) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java:51]: I think it's possible.

**Comment by QinHaiqi** [comment]
> By the way, modify your pr's title.
Thanks for reminding.

**Comment by QinHaiqi** [comment]
> How about reduce your commit number, and detail your pr?
I'll pay attention.

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java (hunk 1 above)
Comment (KevinClair) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java:51]: why?It has been registerd by `@EnableConfigurationProperties(LdapProperties.class)`

Comment (QinHaiqi) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/config/LdapConfiguration.java:51]: I don't understand the function of this annotation. I've learned it now. Thank you.

**Pull request closed without merging**